### PR TITLE
Fix build page when finished.json has 'version' instead of started.json.

### PR DESCRIPTION
**Pull request by rmmh**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1053)

### CODE REVIEW

File: gubernator/main_test.py [hunk 1]
```diff
@@ -48,14 +48,16 @@
 </testsuite>'''
 
 
-def init_build(build_dir, started=True, finished=True):
+def init_build(build_dir, started=True, finished=True,
+               finished_has_version=False):
     """Create faked files for a build."""
+    start_json = {'timestamp': 1406535800}
+    finish_json = {'result': 'SUCCESS', 'timestamp': 1406536800}
+    [start_json, finish_json][finished_has_version]['version'] = 'v1+56'
```
Comment (ixdy) [inline gubernator/main_test.py]: the name for this is changing to `repo-version` and/or `job-version`.

Comment (fejta) [inline gubernator/main_test.py]: What is this python wizardry?

File: gubernator/view_build.py [hunk 2]
```diff
@@ -129,10 +129,12 @@ def get(self, prefix, job, build):
...
             build_log = get_build_log(build_dir)
 
-        if started:
-            commit = started['version'].split('+')[-1]
+        # 'version' might be in either started or finished
+        if started and 'version' in started:
+            version = started['version']
         else:
-            commit = None
+            version = finished and finished.get('version')
```
Comment (fejta) [inline gubernator/view_build.py]: Please favor finished over started

**Comment by rmmh** [comment]
What is `repo-version` and `job-version`? I'll fix that in a separate PR, since 'version' is used pretty heavily.

**Comment by ixdy** [comment]
re: `job-version` https://github.com/kubernetes/kubernetes/pull/36654

**Comment by fejta** [comment]
/lgtm

We should probably wait until everything uses bootstrap to migrate off version.

**Pull request merged**